test/alternator/test_streams.py

```diff
@@ -845,9 +848,11 @@ def test_streams_no_records(test_table_ss_keys_only, dynamodbstreams):
     table, arn = test_table_ss_keys_only
     # Get just one shard - any shard - and its LATEST iterator. Because it's
     # LATEST, there will be no data to read from this iterator.
-    shard_id = dynamodbstreams.describe_stream(StreamArn=arn, Limit=1)['StreamDescription']['Shards'][0]['ShardId']
+    shard = dynamodbstreams.describe_stream(StreamArn=arn, Limit=1)['StreamDescription']['Shards'][0]
+    shard_id = shard['ShardId']
     iter = dynamodbstreams.get_shard_iterator(StreamArn=arn, ShardId=shard_id, ShardIteratorType='LATEST')['ShardIterator']
     response = dynamodbstreams.get_records(ShardIterator=iter)
+    assert 'NextShardIterator' in response or 'EndingSequenceNumber' in shard['SequenceNumberRange']
```

Comment:
This is a weird thing to test here. Nothing this test did closed any shards, in DynamoDB the first case will always be true (there is a NextShardIterator), but in Alternator it depends on whether any closed shards exist - i.e., I expect a 1-node setup to always have the first case, and only a multi-node setup maybe reach the second case. Did you check a multi-node setup?
Moreover, the EndingSequenceNumber test is on the shard, not the iterator. What happens if at the time we got the shard iterator, there it wasn't closed yet, but now, a few seconds later (potentially), it did get closed? In that case, this test will fail - because there is no NextShardIterator, and no EndingSequenceNumber.
I'll try to think how to add in test_streams_closed_read a test for EndingSequenceNumber. It won't help now (this test fails on Alternator) but may help in the future.
In general I  prefer tests which ensure that the situation they are trying to test actually happens. E.g., test_streams_closed_read tried to ensure a closed shard happens - but doesn't work on Alternator.